src/Encryption.php

```diff
@@ -60,7 +60,18 @@ public static function encrypt($payload, $userPublicKey, $userAuthToken, $native
         $userPublicKeyObject = $generator->getPublicKeyFrom($pointUserPublicKey->getX(), $pointUserPublicKey->getY(), $generator->getOrder());
 
         // get shared secret from user public key and local private key
-        $sharedSecret = hex2bin($math->decHex((string) $userPublicKeyObject->getPoint()->mul($localPrivateKeyObject->getSecret())->getX()));
+        $localPrivateSecret = $localPrivateKeyObject->getSecret();
+        
+        $userPublicX = $userPublicKeyObject->getPoint()
```

Comment:
It's always a GMP object, but HHVM may be more strict when attempting casting. I feel that'd be better to gmp_strval all the time instead of casting.